Library/Homebrew/download_strategy.rb

```diff
@@ -282,6 +283,14 @@ def initialize(name, resource)
   end
 
   def fetch
+    if ENV["HOMEBREW_ARTIFACT_DOMAIN"]
+      if @url.include?(BottleSpecification::DEFAULT_DOMAIN)
+        @url.sub!(BottleSpecification::DEFAULT_DOMAIN, ENV["HOMEBREW_ARTIFACT_DOMAIN"])
+      else
+        @url = @url.sub(/^((ht|f)tps?:\/\/)?/, "#{ENV["HOMEBREW_ARTIFACT_DOMAIN"]}/#{Utils::Bottles::Bintray.repository(tap)}/")
```

Comment:
This can also use `@url.sub!`. It also looks really general; can you elaborate on what the input/output URLs should be for this? It'd be even better if you can write a test for both of these cases too.